### ISSUE TITLE
Extending native class with closure referencing this in ctor

### ISSUE THREAD

**Issue by andyli** [HaxeFoundation/haxe#3722]
The following cannot be compiled to Java and C#:

``` haxe
@:nativeGen class A {
    function new():Void {}
}

class Test extends A {
    public function new():Void {
        super();
        var fun = function (){
            this;
        }
    }
    static function main():Void {

    }
}
```

error:

```
Test.hx:6: lines 6-11 : Types that derive from a native class must have its super() call as the first statement in the constructor
```

**Comment by Simn** [comment]
@waneck: Is this a necessary limitation?

**Comment by waneck** [comment]
It is a necessary limitation. We may be able to lift them once bytecode
output is done.
But right now there's no way to lift this limitation for classes that
derive from native classes.

**Comment by andyli** [comment]
What is the exact reason? What could go wrong if super is not called first?

**Comment by ncannasse** [comment]
@andyli I think that's the Java compiler which complains then

**Comment by Simn** [comment]
Yes, my question was if we can transform the problem away. But I suppose we can't because in the case of native super constructors we actually have to call the thing.

Now I'm also worried about the analyzer introducing temp vars above `super()`.

**Comment by Simn** [comment]
Not a regression, moving to 3.3.

**Comment by Simn** [comment]
The Java and C# targets will be put in "regression fixes only" state with the Haxe 4.3.0 release. See https://github.com/HaxeFoundation/haxe/wiki/Target-Support-Level#java-and-c-since-430 for additional information. We apologize for any inconvenience this causes!
